core-tests/shared/src/test/scala/zio/stm/ZSTMSpec.scala

```diff
@@ -414,6 +414,21 @@ object ZSTMSpec extends ZIOBaseSpec {
         } yield assert(result)(equalTo(2 -> 2))
       }
     ),
+    suite("when combinators")(
```

Comment:
Can we also test that the transaction is not executed if the condition is false?

Reply:
I've added them